devicetypes/smartthings/tint-remote-controller.src/tint-remote-controller.groovy

```diff
@@ -0,0 +1,310 @@
+        getOnOffEvent()
+    } else if (descMap.clusterInt == zigbee.COLOR_CONTROL_CLUSTER && descMap.commandInt == MOVE_TO_COLOR_COMMAND) {
+        getColorEvent(descMap)
+    } else if (descMap.clusterInt == zigbee.COLOR_CONTROL_CLUSTER && descMap.commandInt == MOVE_TO_COLOR_TEMPERATURE_COMMAND) {
+		getColorTemperatureEvent(descMap)
+	} else if (descMap.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER && descMap.commandInt == STEP_COMMAND) {
+		getLevelEvent(descMap)
+	} else if (descMap.clusterInt == zigbee.BASIC_CLUSTER && descMap.commandInt == WRITE_ATTRIBUTES_COMMAND) {
+		getButtonEvent(descMap)
+	} else {
+		log.debug "Unhandled message came in, its description map: ${descMap}"
+	}
+}
+
+private isAttrOrCmdMessage(description) {
+    (description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))
+}
+
+private getOnOffEvent() {
+    state.onOffValue = (state.onOffValue == "off" ? "on" : "off")
+    createEvent(name: "switch", value: state.onOffValue)
+}
+
+private getColorEvent(descMap) {
+    def CurrentX = Integer.parseInt(descMap.data[1] + descMap.data[0], 16)
+    def CurrentY = Integer.parseInt(descMap.data[3] + descMap.data[2], 16)
+    def HS = color_xyYtoHSV(CurrentX, CurrentY)
+	return [
+		createEvent(name: "hue", value: HS.H * 100),
+		createEvent(name: "saturation", value: 100)
+	]
+}
+
+private getColorTemperatureEvent(descMap) {
+	def miredTemp = Integer.parseInt(descMap.data[1] + descMap.data[0], 16)
+	def colorTemperature = 1_000_000 / miredTemp
+	colorTemperature = colorTemperature < 2750 ? 2700 : (colorTemperature > 6500 ? 6500 : colorTemperature)
+	return createEvent(name: "colorTemperature", value: colorTemperature)
+}
+
+private getLevelEvent(descMap) {
+	def diff = descMap.data[0] == "00" ? 10 : -10 //0x00 Level up / 0x01 Level down
+	state.currentLevel = state.currentLevel + diff
+	state.currentLevel = Math.min(state.currentLevel, 100)
+	state.currentLevel = Math.max(state.currentLevel, 1)
+	return createEvent(name: "level", value: state.currentLevel)
+}
+
+private getButtonEvent(descMap) {
+	def childId = getButtonNumber(descMap.data[3])
+	def description = "Button no. ${childId} was ${value}"
+	def event = createEvent(name: "button", value: "pushed", descriptionText: description, data: [buttonNumber: childId], isStateChange: true)
+	sendEventToChild(childId, event)
+	return event
+}
+
+def on() {
+	state.onOffValue = "on"
+	createEvent(name: "switch", value: "on")
+}
+
+def off() {
+	state.onOffValue = "off"
+	createEvent(name: "switch", value: "off")
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	return zigbee.readAttribute(zigbee.ONOFF_CLUSTER, ON_OFF_ATTRIBUTE)
+}
+
+def configure() {
+	def groupsList = [0x4004, 0x4005, 0x4006]
+		groupsList.each {
+			sendHubCommand(addHubToGroup(it))
+		}
+	sendEvent(name: "checkInterval", value: (2 * 60 + 10) * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	return zigbee.onOffConfig(5, 3600)
```

Comment:
Do you need to configure color temperature or level?

Reply:
Nah, I've configured on/off just for health check purposes. Which seems to don't work anyway. Any idea what configuration I've could use?